### ISSUE TITLE
Support downgrading

### ISSUE THREAD

**Issue by vincenzovitale** [rt2zz/redux-persist#831]
It would be great to have support for downgrade.
So changing this line:
https://github.com/rt2zz/redux-persist/blob/master/src/createMigrate.js#L33

into something that take a downgrade array of functions and apply a reverse reduce function.

**Comment by arnaudambro** [comment]
I agree, especially when you create branches with new features in the app impacting the state, and that you want to dump afterwards.
